### ISSUE TITLE
Crash with new commit

### ISSUE THREAD

**Issue by weidercs** [evilhero/mylar#1198]
looks like comictagger dies when trying to process after the newest commit:

```
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] dirName:/var/nfs/torrent/complete/comic/
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : ComicTagger Path location for internal comictagger.py set to : /home/xxx/Apps/Mylar/comictagger.py
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] UNRAR path set to : /usr/bin/unrar
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Filename is : /var/nfs/torrent/complete/comic/Itty Bitty Hellboy - The Search for the Were-Jaguar! 03 (of 04) (2016) (digital) (Son of Ultron-Empire).cbr
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Paths / Locations:
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] scriptname : Mylar.py
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] downloadpath : /var/nfs/torrent/complete/comic
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] sabnzbdscriptpath : 
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] comicpath : /home/xxx/Apps/Mylar/cache/mylar_TuB1cd
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] unrar_folder : /home/xxx/Apps/Mylar/cache/mylar_TuB1cd/unrard
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Running the ComicTagger Add-on for Mylar
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] ComicTagger 1.1.15-beta being used - using personal ComicVine API key supplied via mylar.
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] CR Tagging enabled.
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] CBL Tagging enabled.
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Enabling ComicTagger script: /usr/bin/python /home/xxx/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_TuB1cd/Itty Bitty Hellboy - The Search for the Were-Jaguar! 03 (of 04) (2016) (digital) (Son of Ultron-Empire).cbr']
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Executing command: ['/usr/bin/python', '/home/xxx/Apps/Mylar/comictagger.py', '-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_TuB1cd/Itty Bitty Hellboy - The Search for the Were-Jaguar! 03 (of 04) (2016) (digital) (Son of Ultron-Empire).cbr']
27-Jan-2016 08:33:39 - DEBUG   :: Thread-12 : [META-TAGGER] Absolute path to script: /usr/bin/python
27-Jan-2016 08:33:39 - ERROR   :: Thread-12 : Uncaught exception: Traceback (most recent call last):
  File "/home/xxx/Apps/Mylar/mylar/logger.py", line 158, in new_run
    old_run(*args, **kwargs)
  File "/usr/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/xxx/Apps/Mylar/mylar/PostProcessor.py", line 931, in Process
    self.Process_next(comicid, issueid, issuenumOG, ml, stat)
  File "/home/xxx/Apps/Mylar/mylar/PostProcessor.py", line 1215, in Process_next
    pcheck = cmtagmylar.run(self.nzb_folder, issueid=issueid, comversion=comversion, manual="yes", filename=ml['ComicLocation'])
  File "/home/xxx/Apps/Mylar/mylar/cmtagmylar.py", line 250, in run
    logger.info(module + '[COMIC-TAGGER] Successfully wrote ' + tagdisp + ' [' + filepath + ']')
UnboundLocalError: local variable 'tagdisp' referenced before assignment

```

**Comment by weidercs** [comment]
I commented out the use of tagdisp on line 250 in cmtagmylar.py:

```
            else:
                #logger.info(module + '[COMIC-TAGGER] Successfully wrote ' + tagdisp + ' [' + filepath + ']')
                i+=1
```
and it seems to work.  I'm sure it's a band-aid, but got me working....

**Comment by evilhero** [comment]
I was getting this abit last night prior to the last commit going out - and had it patched/fixed so it wasn't doing it anymore.

I'm curious if this has reared it's head again for me- is this happening to every comic you attempt to meta-tag? I'm also guessing this is for manual meta-tagging ?

Can you confirm the commit you're currently running as well - there were some upgrade problems in the previous commit. If it fails to update for some reason, you would have to delete the contents of your lib/comictaggerlib/ , but keep the folder structure of lib/comictaggerlib/ intact.

**Comment by weidercs** [comment]
Mylar Version: development
-- git build 52d0b29702359c37352ed2194de05fa8db201b8f.
Python Version : 2.7.10

It was happening for every comic, seems smooth once I commented out that line.

**Comment by evilhero** [comment]
When I was seeing that error prior to the commit going out, the cbz's were being generated but the tagging wasn't occuring. So you'd have a .cbz file but no tagging within said cbz file.

Can you confirm for me if there's tagging within the cbz file (whether it's the xml file or the zip comments)

**Comment by weidercs** [comment]
ok, I'll revert that file and see what I can find out...

**Comment by weidercs** [comment]
no tags written, still a cbr... looks like it moved the cbr to a cache diretory, then crapped out...
```
xxx@media1:~/Apps/Mylar/cache/mylar_tOKGIi$ ls
Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
xxx@media1:~/Apps/Mylar/cache/mylar_tOKGIi$ pwd
/home/xxx/Apps/Mylar/cache/mylar_tOKGIi
```

All done automatically no manual intervention:

```
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] dirName:/var/nfs/torrent/complete/comic/
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : ComicTagger Path location for internal comictagger.py set to : /home/xxx/Apps/Mylar/comictagger.py
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] UNRAR path set to : /usr/bin/unrar
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Filename is : /var/nfs/torrent/complete/comic/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Paths / Locations:
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] scriptname : Mylar.py
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] downloadpath : /var/nfs/torrent/complete/comic
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] sabnzbdscriptpath : 
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] comicpath : /home/xxx/Apps/Mylar/cache/mylar_tOKGIi
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] unrar_folder : /home/xxx/Apps/Mylar/cache/mylar_tOKGIi/unrard
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Running the ComicTagger Add-on for Mylar
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] ComicTagger 1.1.15-beta being used - using personal ComicVine API key supplied via mylar.
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] CR Tagging enabled.
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] CBL Tagging enabled.
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Enabling ComicTagger script: /usr/bin/python /home/xxx/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_tOKGIi/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Executing command: ['/usr/bin/python', '/home/xxx/Apps/Mylar/comictagger.py', '-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_tOKGIi/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Absolute path to script: /usr/bin/python
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : Exception generated in thread FOLDERMONITOR: local variable 'tagdisp' referenced before assignment
27-Jan-2016 10:56:47 - DEBUG   :: FOLDERMONITOR : 'Traceback (most recent call last):\n  File "/home/xxx/Apps/Mylar/mylar/scheduler.py", line 78, in runAction\n    self.action.run()\n  File "/home/xxx/Apps/Mylar/mylar/PostProcessor.py", line 1681, in run\n    result = PostProcess.Process()\n  File "/home/xxx/Apps/Mylar/mylar/PostProcessor.py", line 931, in Process\n    self.Process_next(comicid, issueid, issuenumOG, ml, stat)\n  File "/home/xxx/Apps/Mylar/mylar/PostProcessor.py", line 1215, in Process_next\n    pcheck = cmtagmylar.run(self.nzb_folder, issueid=issueid, comversion=comversion, manual="yes", filename=ml[\'ComicLocation\'])\n  File "/home/xxx/Apps/Mylar/mylar/cmtagmylar.py", line 250, in run\n    logger.info(module + \'[COMIC-TAGGER] Successfully wrote \' + tagdisp + \' [\' + filepath + \']\')\nUnboundLocalError: local variable \'tagdisp\' referenced before assignment\n'
```

**Comment by weidercs** [comment]
And after I comment out line 250 in cmtagmylar.py it runs smooooooth:
```
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] dirName:/var/nfs/torrent/complete/comic/
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : ComicTagger Path location for internal comictagger.py set to : /home/xxx/Apps/Mylar/comictagger.py
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] UNRAR path set to : /usr/bin/unrar
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Filename is : /var/nfs/torrent/complete/comic/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Paths / Locations:
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] scriptname : Mylar.py
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] downloadpath : /var/nfs/torrent/complete/comic
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] sabnzbdscriptpath : 
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] comicpath : /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] unrar_folder : /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/unrard
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Running the ComicTagger Add-on for Mylar
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] ComicTagger 1.1.15-beta being used - using personal ComicVine API key supplied via mylar.
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] CR Tagging enabled.
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] CBL Tagging enabled.
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Enabling ComicTagger script: /usr/bin/python /home/xxx/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Executing command: ['/usr/bin/python', '/home/xxx/Apps/Mylar/comictagger.py', '-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Absolute path to script: /usr/bin/python
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Enabling ComicTagger script: /usr/bin/python /home/xxx/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Executing command: ['/usr/bin/python', '/home/xxx/Apps/Mylar/comictagger.py', '-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']
27-Jan-2016 11:13:30 - DEBUG   :: FOLDERMONITOR : [META-TAGGER] Absolute path to script: /usr/bin/python
27-Jan-2016 11:13:31 - INFO    :: FOLDERMONITOR : [POST-PROCESSING] Sucessfully wrote metadata to .cbz (Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr) - Continuing..
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] ofilename:Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] odir: /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] ofilename: Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] ext:.cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Original Filename: Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Original Extension: .cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] New Filename: Old.Man.Logan.001.(2016)
27-Jan-2016 11:13:31 - INFO    :: FOLDERMONITOR : [POST-PROCESSING][DIRECTORY-CHECK] Found comic directory: /mnt/Media/Comics/Old.Man.Logan.(2016)
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Source: /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Destination: /mnt/Media/Comics/Old.Man.Logan.(2016)/Old.Man.Logan.001.(2016).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Renaming /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] odir src : /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] moveing /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr ... to ... /mnt/Media/Comics/Old.Man.Logan.(2016)/Old.Man.Logan.001.(2016).cbr
27-Jan-2016 11:13:31 - INFO    :: FOLDERMONITOR : [POST-PROCESSING] move successful to : /mnt/Media/Comics/Old.Man.Logan.(2016)/Old.Man.Logan.001.(2016).cbr
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING]self.nzb_folder: /var/nfs/torrent/complete/comic/
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING] Tidying up. Deleting folder : /home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : Successfully changed permissions [0777 / 0660]
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING][UPDATER] comicid: 87624
27-Jan-2016 11:13:31 - DEBUG   :: FOLDERMONITOR : [POST-PROCESSING][UPDATER] issueid: 512432
27-Jan-2016 11:13:31 - INFO    :: FOLDERMONITOR : [POST-PROCESSING][UPDATER] Setting status to Post-Processed in history.
27-Jan-2016 11:13:31 - INFO    :: FOLDERMONITOR : [POST-PROCESSING][UPDATER] Updating Status (Post-Processed) now complete for Old Man Logan issue: 1
```

**Comment by weidercs** [comment]
Now that I look at it closer, I take that back, it stays as a cbr, no tagging is done. Weird.

**Comment by evilhero** [comment]
Oh crap - I can see where it's messing up in that last log file.
```
Enabling ComicTagger script: /usr/bin/python /home/xxx/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr', u'/home/xxx/Apps/Mylar/cache/mylar_wPZ7Ta/Old Man Logan 001 (2016) (digital) (Oroboros-DCP).cbr']```

It's submitting 2 files for conversion/change and it's only expecting one so it's crapping out. 

I'm betting the cbz file doesn't have any tags within it.

**Comment by weidercs** [comment]
correct, also not renamed to cbz, it stays cbr.  Just tried manually tagging and it does the same non-action.

**Comment by evilhero** [comment]
I''m wondering if your comictaggerlib folder got updated properly...it doesn't seem to be using the proper changes from the files that got pushed out in the last commit or two.

**Comment by weidercs** [comment]
I'll do some brute force git reset action and test that...

**Comment by evilhero** [comment]
Thanks, I'm working through some of the code on this end just to see if there's something amiss with the tagging options, so far no luck though...

**Comment by weidercs** [comment]
something new after I did:
git fetch --all
git reset --hard origin/development

when I try to manually tag I get:
```
27-Jan-2016 11:37:20 - WARNING :: CP Server Thread-4 : [META-TAGGER] Unable to create temporary directory to perform meta-tagging. Processing without metatagging.
```

where does it try to create a temporary directory? nothing else was in the logs

**Comment by evilhero** [comment]
It creates the temporary directories within the cache folder

**Comment by evilhero** [comment]
I just tried meta-tagging on mine again...
```
[META-TAGGER] Enabling ComicTagger script: python /mylar/comictagger.py with options: ['-s', '-m', 'volume=', '--cv-api-key', 'XXXXXXXXX'', '-o', '--id', u'512471', '--type', 'cr', '/mylar/cache/mylar_jW9vY2/Saga 033 (2016).cbz']
```
Tagged and worked ok - no double calls within for some reason ...

**Comment by weidercs** [comment]
I did a complete re-install and just copied over my config.ini and mylar.db files.  My tagging script still doesn't look like yours:
```
[META-TAGGER] Enabling ComicTagger script: /usr/bin/python /home/weidercs/Apps/Mylar/comictagger.py with options: ['-e', '--delete-rar', u'/home/weidercs/Apps/Mylar/cache/mylar_Wr6sZA/Venom.Space.Knight.002.(2016).cbr']
```
only calling once, but it doesn't have the other options, also does nothing but move the file to cache directory (no tagging done)

Thoughts? Something bad in my config.ini? this is how it's running (added the -v to test):
```
/usr/bin/python Mylar.py -v --daemon --nolaunch --pidfile=/var/run/mylar/mylar.pid --datadir=/home/xxx/Apps/Mylar --port=8090
```

**Comment by evilhero** [comment]
You don't have to specify the port & datadir options as those are the default values that you're using anyways.

Have you tried running mylar directly from the cli in non-daemon mode? It's possible it's having an issue with the stdout for reason, which might explain some things but leave question marks about others.

**Comment by weidercs** [comment]
Tried cli with just the -v --nolaunch options, no dice.

I fought this for quite a while and still wasn't successful, I rolled back to master branch and ended up with an issue with that branch until I added the cmtagger_path location back to the config.ini file... So I'm up and working now, but just not on the development branch.  I know how to switch back and forth pretty easy now, so if you want me to do more debugging, let me know.

**Comment by evilhero** [comment]
Figured this out - this is an actual bug with the 5.3 branch of unrar. It's been documented and patched by the sickrage folks [here](https://github.com/yk4ever/py-unrar2/pull/15), but it hasn't actually merged into the unrar module as of yet.

I've updated the py-unrar2 module that's already in mylar to the latest version and then diff applied the patch so it's all up to date and spanky again (thanks to the sickRage team).

Will be in the next development commit coming shortly...
